### PR TITLE
refactor aad sample code

### PR DESCRIPTION
**Pull request by backwind1233**
- make the sample code clean and easier to config

### CODE REVIEW

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md [hunk 1]
```diff
@@ -87,57 +87,53 @@ As a first step you'll need to:
...
 
-In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+In the steps below, "client-id" is the same as "Application ID" or "AppId".
+
+Open application.yml in your project to configure:
+
+```yml
+azure:
+  activedirectory:
+    tenantId: <your-tenantId>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md:97]: we'd better use the camel case here for all properties to make them consistent.

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md [hunk 2]
```diff
@@ -87,57 +87,53 @@ As a first step you'll need to:
...
+      allowed-groups: group1,group2
+    # Optional, the default value is 
+    # environment: global  
+```
+
 
-Open application.properties in your project to configure:
 
-1. If your azure account follows format xxx@xxx.partner.onmschina.cn, configure property `azure.activedirectory.environment=cn` to use [Azure China](https://docs.microsoft.com/azure/china/china-welcome), the default value is `global`.
+1. If your azure account follows format xxx@xxx.partner.onmschina.cn, configure property `environment: cn` to use [Azure China](https://docs.microsoft.com/azure/china/china-welcome), the default value is `global`.
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md:111]: Let's use the reference link style here, and put all our links at the bottom of this README file for better maintenance.

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md [hunk 3]
```diff
@@ -87,57 +87,53 @@ As a first step you'll need to:
...
 
 2. Put Application ID and client-secret in `client-id` and `client-secret` respectively e.g.
 ```properties
-azure.activedirectory.client-id=xxxxxx-your-client-id-xxxxxx
-azure.activedirectory.client-secret=xxxxxx-your-client-secret-xxxxxx
+tenantId: xxxxxx-your-client-id-xxxxxx
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md:115]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md [hunk 4]
```diff
@@ -87,57 +87,53 @@ As a first step you'll need to:
...
+Open application.yml in your project to configure:
+
+```yml
+azure:
+  activedirectory:
+    tenantId: <your-tenantId>
+    client-id: <your-client-id>
+    client-secret: <your-client-secret>
+    # Optional, default value is http://localhost:8080/
+    redirectUriTemplate: <your-redirect-uri>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md:101]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md [hunk 5]
```diff
@@ -87,57 +87,53 @@ As a first step you'll need to:
...
 ```properties
 # groups that you created in your Azure AD tenant
-azure.activedirectory.user-group.allowed-groups=group1,group2
+allowed-groups: group1,group2
 ```
 
 4. (Optional) If you want to configure oauth2 redirect uri, please configure by :
 ```properties
-spring.security.oauth2.client.registration.azure.redirect-uri=xxxxxx-your-redirect-uri-xxxxxx
+redirectUriTemplate: xxxxxx-your-redirect-uri-xxxxxx
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md:127]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md [hunk 6]
```diff
@@ -87,57 +87,53 @@ As a first step you'll need to:
...
 ```
 
- ---
+---
  ### Step 4: Change Role_group1 to your group
 1. You can use `@PreAuthorize` annotation or `UserPrincipal` to manage access to web API based on user's group membership. You will need to change `ROLE_group1` to groups you want to allow to access the API in `TodoListController.java` or you will get "Access is denied".
    
----   
-### Step 5: Angular JS
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/README.md:118]: awesome

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/java/com/azure/spring/sample/aad/controller/TodoListController.java [hunk 7]
```diff
@@ -45,6 +47,18 @@ public TodoListController() {
         return model;
     }
 
+
+    @RequestMapping({"/"})
+    public ModelAndView index() {
+        ModelAndView model = new ModelAndView("index");
+        model.addObject("aad_clientId", aadAuthenticationProperties.getClientId());
+        model.addObject("aad_tenantId", aadAuthenticationProperties.getTenantId());
+        model.addObject("aad_redirectUri", Optional
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/java/com/azure/spring/sample/aad/controller/TodoListController.java:56]: code style?

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/resources/application.yml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# By default, azure.activedirectory.environment property has value `global`,
+# supported value is global, cn. Please refer to the README for details.
+# azure.activedirectory.environment=global
+azure:
+  activedirectory:
+    tenantId: <your-tenantId>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/resources/application.yml:6]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/resources/application.yml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    client-id: <your-client-id>
+    client-secret: <your-client-secret>
+    # Optional, default value is http://localhost:8080/
+    # redirectUriTemplate: <your-redirect-uri>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/resources/application.yml:10]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/resources/static/App/Scripts/app.js [hunk 10]
```diff
@@ -15,15 +15,14 @@ angular.module('todoApp', ['ngRoute', 'MsalAngular'])
         }).otherwise({redirectTo: "/Home"});
 
         window.applicationConfig = {
-            clientID: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
+            clientID: window.aad_clientId
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/resources/static/App/Scripts/app.js:18]: wonderful

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/resources/templates/index.html [hunk 11]
```diff
@@ -1,10 +1,19 @@
 ﻿<!DOCTYPE html>
-<html xmlns="http://www.w3.org/1999/xhtml">
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:th="http://www.thymeleaf.org">
+
 <head>
     <title>Todo List: a SPA sample demonstrating Azure AD and MSAL JS</title>
     <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css">
     <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap-theme.min.css">
 
+    <script th:inline="javascript">
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/resources/templates/index.html:9]: Good catch.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/resources/static/App/Scripts/app.js (hunk 10 above)
Comment (backwind1233) [reply sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/resources/static/App/Scripts/app.js]: unbelievable

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/java/com/azure/spring/sample/aad/controller/TodoListController.java (hunk 7 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-by-filter/src/main/java/com/azure/spring/sample/aad/controller/TodoListController.java:56]: I think @saragluna  means change `aad_clientId` to `aad_client_id`.

**Review by saragluna — APPROVED** [review]

**Pull request merged**